[
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1172582444",
        "html_url": "https://github.com/rapidsai/cudf/issues/11189#issuecomment-1172582444",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/11189",
        "id": 1172582444,
        "node_id": "IC_kwDOBWUGps5F5DAs",
        "user": {
            "login": "VibhuJawa",
            "id": 4837571,
            "node_id": "MDQ6VXNlcjQ4Mzc1NzE=",
            "avatar_url": "https://avatars.githubusercontent.com/u/4837571?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/VibhuJawa",
            "html_url": "https://github.com/VibhuJawa",
            "followers_url": "https://api.github.com/users/VibhuJawa/followers",
            "following_url": "https://api.github.com/users/VibhuJawa/following{/other_user}",
            "gists_url": "https://api.github.com/users/VibhuJawa/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/VibhuJawa/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/VibhuJawa/subscriptions",
            "organizations_url": "https://api.github.com/users/VibhuJawa/orgs",
            "repos_url": "https://api.github.com/users/VibhuJawa/repos",
            "events_url": "https://api.github.com/users/VibhuJawa/events{/privacy}",
            "received_events_url": "https://api.github.com/users/VibhuJawa/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2022-07-01T17:53:16Z",
        "updated_at": "2022-07-01T18:00:25Z",
        "author_association": "MEMBER",
        "body": "Thanks for triaging and raising this @ChrisJar .  I can confirm that i can reproduce this on my end too. \r\n\r\nSmaller minimal example for anyone interested. \r\n\r\n```python\r\nimport numpy as np\r\nimport cudf\r\nimport dask_cudf\r\n\r\ndf_a = cudf.DataFrame({'a':[1, 2, 3, 4, 5], 'b':[0]*5})\r\ndf_a['a']=df_a['a'].astype(np.int32)\r\n\r\ndf_b = cudf.DataFrame({'a':[5, 4, 3, 2, 1], 'c':[1]*5})\r\ndf_b['a']=df_b['a'].astype(np.int64)\r\nprint(\"cudf\\n\",df_a.merge(df_b))\r\nprint(\"--\"*10)\r\nddf_a = dask_cudf.from_cudf(df_a, npartitions=4)\r\nddf_b = dask_cudf.from_cudf(df_b, npartitions=4)\r\nprint(\"dask_cudf\\n\",ddf_a.merge(ddf_b).compute())\r\n```\r\n\r\n```\r\ncudf\r\n    a  b  c\r\n0  1  0  1\r\n1  2  0  1\r\n2  3  0  1\r\n3  4  0  1\r\n4  5  0  1\r\n--------------------\r\ndask_cudf\r\n    a  b  c\r\n0  3  0  1\r\n0  1  0  1\r\n```\r\n\r\n\r\nAlso note that this work with `dask_pandas` .\r\n```python\r\n\r\nimport numpy as np\r\nimport pandas as pd\r\nimport dask.dataframe\r\n\r\ndf_a = pd.DataFrame({'a':[1, 2, 3, 4, 5], 'b':[0]*5})\r\ndf_a['a']=df_a['a'].astype(np.int32)\r\n\r\ndf_b = pd.DataFrame({'a':[5, 4, 3, 2, 1], 'c':[1]*5})\r\ndf_b['a']=df_b['a'].astype(np.int64)\r\nprint(\"cudf\\n\",df_a.merge(df_b))\r\nprint(\"--\"*10)\r\nddf_a = dask.dataframe.from_pandas(df_a, npartitions=4)\r\nddf_b = dask.dataframe.from_pandas(df_b, npartitions=4)\r\nprint(\"dask_cudf\\n\",ddf_a.merge(ddf_b).compute())\r\n``` \r\n\r\n```\r\ncudf\r\n    a  b  c\r\n0  1  0  1\r\n1  2  0  1\r\n2  3  0  1\r\n3  4  0  1\r\n4  5  0  1\r\n--------------------\r\ndask_cudf\r\n    a  b  c\r\n0  1  0  1\r\n0  3  0  1\r\n1  4  0  1\r\n2  5  0  1\r\n3  2  0  1\r\n```\r\n",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1172582444/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1172593338",
        "html_url": "https://github.com/rapidsai/cudf/issues/11189#issuecomment-1172593338",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/11189",
        "id": 1172593338,
        "node_id": "IC_kwDOBWUGps5F5Fq6",
        "user": {
            "login": "VibhuJawa",
            "id": 4837571,
            "node_id": "MDQ6VXNlcjQ4Mzc1NzE=",
            "avatar_url": "https://avatars.githubusercontent.com/u/4837571?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/VibhuJawa",
            "html_url": "https://github.com/VibhuJawa",
            "followers_url": "https://api.github.com/users/VibhuJawa/followers",
            "following_url": "https://api.github.com/users/VibhuJawa/following{/other_user}",
            "gists_url": "https://api.github.com/users/VibhuJawa/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/VibhuJawa/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/VibhuJawa/subscriptions",
            "organizations_url": "https://api.github.com/users/VibhuJawa/orgs",
            "repos_url": "https://api.github.com/users/VibhuJawa/repos",
            "events_url": "https://api.github.com/users/VibhuJawa/events{/privacy}",
            "received_events_url": "https://api.github.com/users/VibhuJawa/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2022-07-01T18:06:26Z",
        "updated_at": "2022-07-01T18:06:26Z",
        "author_association": "MEMBER",
        "body": "CC: @rjzamora , @galipremsagar ",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1172593338/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1172628669",
        "html_url": "https://github.com/rapidsai/cudf/issues/11189#issuecomment-1172628669",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/11189",
        "id": 1172628669,
        "node_id": "IC_kwDOBWUGps5F5OS9",
        "user": {
            "login": "rjzamora",
            "id": 20461013,
            "node_id": "MDQ6VXNlcjIwNDYxMDEz",
            "avatar_url": "https://avatars.githubusercontent.com/u/20461013?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/rjzamora",
            "html_url": "https://github.com/rjzamora",
            "followers_url": "https://api.github.com/users/rjzamora/followers",
            "following_url": "https://api.github.com/users/rjzamora/following{/other_user}",
            "gists_url": "https://api.github.com/users/rjzamora/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/rjzamora/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/rjzamora/subscriptions",
            "organizations_url": "https://api.github.com/users/rjzamora/orgs",
            "repos_url": "https://api.github.com/users/rjzamora/repos",
            "events_url": "https://api.github.com/users/rjzamora/events{/privacy}",
            "received_events_url": "https://api.github.com/users/rjzamora/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2022-07-01T18:55:11Z",
        "updated_at": "2022-07-01T19:23:50Z",
        "author_association": "MEMBER",
        "body": "My immediate assumption is that \"int32\" and \"int64\" columns are hashing to different partitions, but I'll need to investigate a bit to say for sure.\r\n\r\n**EDIT**: I can confirm that `pd.util.hash_pandas_object` produces the same hash for \"int32\" and \"int64\" values, while `cudf.DataFrame.hash_values()` does not:\r\n\r\n```python\r\ncudf_a = cudf.DataFrame({\"a\": np.arange(10, dtype=\"int32\")}).hash_values()\r\ncudf_b = cudf.DataFrame({\"a\": np.arange(10, dtype=\"int64\")}).hash_values()\r\nassert all(cudf_a.values == cudf_b.values)  # FAILS\r\n\r\npd_a = pd.util.hash_pandas_object(pd.DataFrame({\"a\": np.arange(10, dtype=\"int32\")}))\r\npd_b = pd.util.hash_pandas_object(pd.DataFrame({\"a\": np.arange(10, dtype=\"int64\")}))\r\nassert all(pd_a.values == pd_b.values) # PASSES\r\n```",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1172628669/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1175201725",
        "html_url": "https://github.com/rapidsai/cudf/issues/11189#issuecomment-1175201725",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/11189",
        "id": 1175201725,
        "node_id": "IC_kwDOBWUGps5GDCe9",
        "user": {
            "login": "rjzamora",
            "id": 20461013,
            "node_id": "MDQ6VXNlcjIwNDYxMDEz",
            "avatar_url": "https://avatars.githubusercontent.com/u/20461013?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/rjzamora",
            "html_url": "https://github.com/rjzamora",
            "followers_url": "https://api.github.com/users/rjzamora/followers",
            "following_url": "https://api.github.com/users/rjzamora/following{/other_user}",
            "gists_url": "https://api.github.com/users/rjzamora/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/rjzamora/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/rjzamora/subscriptions",
            "organizations_url": "https://api.github.com/users/rjzamora/orgs",
            "repos_url": "https://api.github.com/users/rjzamora/repos",
            "events_url": "https://api.github.com/users/rjzamora/events{/privacy}",
            "received_events_url": "https://api.github.com/users/rjzamora/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2022-07-05T15:35:49Z",
        "updated_at": "2022-07-05T15:43:28Z",
        "author_association": "MEMBER",
        "body": "@ChrisJar @VibhuJawa - I just noticed that neither of you have mentioned the `UserWarning` you should get for the reproducers shared above. For example, I get the following message locally:\r\n\r\n```\r\n.../site-packages/dask/dataframe/multi.py:475: UserWarning: Merging dataframes with merge column data type mismatches: \r\n+---------------+------------+-------------+\r\n| Merge columns | left dtype | right dtype |\r\n+---------------+------------+-------------+\r\n| ('a', 'a')    | int32      | int64       |\r\n+---------------+------------+-------------+\r\nCast dtypes explicitly to avoid unexpected results.\r\n```\r\n\r\nI do think it makes sense to cast the columns automatically (when possible), but I just wanted to confirm that you are gettting this warning as well?",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1175201725/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1175203271",
        "html_url": "https://github.com/rapidsai/cudf/issues/11189#issuecomment-1175203271",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/11189",
        "id": 1175203271,
        "node_id": "IC_kwDOBWUGps5GDC3H",
        "user": {
            "login": "galipremsagar",
            "id": 11664259,
            "node_id": "MDQ6VXNlcjExNjY0MjU5",
            "avatar_url": "https://avatars.githubusercontent.com/u/11664259?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/galipremsagar",
            "html_url": "https://github.com/galipremsagar",
            "followers_url": "https://api.github.com/users/galipremsagar/followers",
            "following_url": "https://api.github.com/users/galipremsagar/following{/other_user}",
            "gists_url": "https://api.github.com/users/galipremsagar/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/galipremsagar/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/galipremsagar/subscriptions",
            "organizations_url": "https://api.github.com/users/galipremsagar/orgs",
            "repos_url": "https://api.github.com/users/galipremsagar/repos",
            "events_url": "https://api.github.com/users/galipremsagar/events{/privacy}",
            "received_events_url": "https://api.github.com/users/galipremsagar/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2022-07-05T15:37:20Z",
        "updated_at": "2022-07-05T15:37:59Z",
        "author_association": "CONTRIBUTOR",
        "body": "Yup, I did get that warning @rjzamora. ",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1175203271/reactions",
            "total_count": 1,
            "+1": 1,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1175209713",
        "html_url": "https://github.com/rapidsai/cudf/issues/11189#issuecomment-1175209713",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/11189",
        "id": 1175209713,
        "node_id": "IC_kwDOBWUGps5GDEbx",
        "user": {
            "login": "ChrisJar",
            "id": 13131098,
            "node_id": "MDQ6VXNlcjEzMTMxMDk4",
            "avatar_url": "https://avatars.githubusercontent.com/u/13131098?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/ChrisJar",
            "html_url": "https://github.com/ChrisJar",
            "followers_url": "https://api.github.com/users/ChrisJar/followers",
            "following_url": "https://api.github.com/users/ChrisJar/following{/other_user}",
            "gists_url": "https://api.github.com/users/ChrisJar/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/ChrisJar/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/ChrisJar/subscriptions",
            "organizations_url": "https://api.github.com/users/ChrisJar/orgs",
            "repos_url": "https://api.github.com/users/ChrisJar/repos",
            "events_url": "https://api.github.com/users/ChrisJar/events{/privacy}",
            "received_events_url": "https://api.github.com/users/ChrisJar/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2022-07-05T15:43:36Z",
        "updated_at": "2022-07-05T15:43:36Z",
        "author_association": "CONTRIBUTOR",
        "body": "Yup, so did I @rjzamora ",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1175209713/reactions",
            "total_count": 1,
            "+1": 1,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1176389470",
        "html_url": "https://github.com/rapidsai/cudf/issues/11189#issuecomment-1176389470",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/11189",
        "id": 1176389470,
        "node_id": "IC_kwDOBWUGps5GHkde",
        "user": {
            "login": "VibhuJawa",
            "id": 4837571,
            "node_id": "MDQ6VXNlcjQ4Mzc1NzE=",
            "avatar_url": "https://avatars.githubusercontent.com/u/4837571?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/VibhuJawa",
            "html_url": "https://github.com/VibhuJawa",
            "followers_url": "https://api.github.com/users/VibhuJawa/followers",
            "following_url": "https://api.github.com/users/VibhuJawa/following{/other_user}",
            "gists_url": "https://api.github.com/users/VibhuJawa/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/VibhuJawa/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/VibhuJawa/subscriptions",
            "organizations_url": "https://api.github.com/users/VibhuJawa/orgs",
            "repos_url": "https://api.github.com/users/VibhuJawa/repos",
            "events_url": "https://api.github.com/users/VibhuJawa/events{/privacy}",
            "received_events_url": "https://api.github.com/users/VibhuJawa/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2022-07-06T15:51:09Z",
        "updated_at": "2022-07-06T15:51:09Z",
        "author_association": "MEMBER",
        "body": "@ChrisJar , Can you raise an issue on `dask-sql`  too just as it will be hard for users to work-around from that world as there the merge can be present deep inside a messy query.  ",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1176389470/reactions",
            "total_count": 1,
            "+1": 1,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1176440489",
        "html_url": "https://github.com/rapidsai/cudf/issues/11189#issuecomment-1176440489",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/11189",
        "id": 1176440489,
        "node_id": "IC_kwDOBWUGps5GHw6p",
        "user": {
            "login": "rjzamora",
            "id": 20461013,
            "node_id": "MDQ6VXNlcjIwNDYxMDEz",
            "avatar_url": "https://avatars.githubusercontent.com/u/20461013?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/rjzamora",
            "html_url": "https://github.com/rjzamora",
            "followers_url": "https://api.github.com/users/rjzamora/followers",
            "following_url": "https://api.github.com/users/rjzamora/following{/other_user}",
            "gists_url": "https://api.github.com/users/rjzamora/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/rjzamora/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/rjzamora/subscriptions",
            "organizations_url": "https://api.github.com/users/rjzamora/orgs",
            "repos_url": "https://api.github.com/users/rjzamora/repos",
            "events_url": "https://api.github.com/users/rjzamora/events{/privacy}",
            "received_events_url": "https://api.github.com/users/rjzamora/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2022-07-06T16:39:10Z",
        "updated_at": "2022-07-06T16:39:10Z",
        "author_association": "MEMBER",
        "body": "@VibhuJawa - Do you have a sense for what kinds of dtype mismatch we should worry about from dask-sql.  It seems like we should be able to handle mismatched integer dtypes in Dask, but I wasn't planning on addressing anything else since the user should be getting a warning that their doing something dangerous/wrong (and \"general\" type-casting rules are pretty vague). However, you make a good point that the user may not have much control if they are working with the dask-sql API.",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1176440489/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1176458904",
        "html_url": "https://github.com/rapidsai/cudf/issues/11189#issuecomment-1176458904",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/11189",
        "id": 1176458904,
        "node_id": "IC_kwDOBWUGps5GH1aY",
        "user": {
            "login": "VibhuJawa",
            "id": 4837571,
            "node_id": "MDQ6VXNlcjQ4Mzc1NzE=",
            "avatar_url": "https://avatars.githubusercontent.com/u/4837571?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/VibhuJawa",
            "html_url": "https://github.com/VibhuJawa",
            "followers_url": "https://api.github.com/users/VibhuJawa/followers",
            "following_url": "https://api.github.com/users/VibhuJawa/following{/other_user}",
            "gists_url": "https://api.github.com/users/VibhuJawa/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/VibhuJawa/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/VibhuJawa/subscriptions",
            "organizations_url": "https://api.github.com/users/VibhuJawa/orgs",
            "repos_url": "https://api.github.com/users/VibhuJawa/repos",
            "events_url": "https://api.github.com/users/VibhuJawa/events{/privacy}",
            "received_events_url": "https://api.github.com/users/VibhuJawa/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2022-07-06T16:59:00Z",
        "updated_at": "2022-07-06T17:04:25Z",
        "author_association": "MEMBER",
        "body": "@rjzamora ,   I agree that  \"general\" type-casting rules are pretty vague so having a general agree-able behavior will be impossible. \r\n\r\n> What type of  dtype mismatch we should worry about from dask-sql.\r\n\r\nIn my opinion  we should handle merges b/w columns of similar dtype , say like `int32/int64` and `float32/float64` etc and FAIL loudly in other cases (like `int32`/`float32`,  `int32/uint32`  )  for `dask-sql`  .  \r\n\r\nI would love for @ayushdg  to weigh in on this as he is deep in the weeds for handling merging on the `dask-sql` side these days. \r\n\r\nJust for additional context, this issue was actually found when we were triaging incorrect results in a GPU-BDB dask-sql query   (See [issue](https://github.com/rapidsai/gpu-bdb/issues/262))\r\n",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1176458904/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1176467159",
        "html_url": "https://github.com/rapidsai/cudf/issues/11189#issuecomment-1176467159",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/11189",
        "id": 1176467159,
        "node_id": "IC_kwDOBWUGps5GH3bX",
        "user": {
            "login": "rjzamora",
            "id": 20461013,
            "node_id": "MDQ6VXNlcjIwNDYxMDEz",
            "avatar_url": "https://avatars.githubusercontent.com/u/20461013?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/rjzamora",
            "html_url": "https://github.com/rjzamora",
            "followers_url": "https://api.github.com/users/rjzamora/followers",
            "following_url": "https://api.github.com/users/rjzamora/following{/other_user}",
            "gists_url": "https://api.github.com/users/rjzamora/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/rjzamora/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/rjzamora/subscriptions",
            "organizations_url": "https://api.github.com/users/rjzamora/orgs",
            "repos_url": "https://api.github.com/users/rjzamora/repos",
            "events_url": "https://api.github.com/users/rjzamora/events{/privacy}",
            "received_events_url": "https://api.github.com/users/rjzamora/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2022-07-06T17:04:31Z",
        "updated_at": "2022-07-06T17:04:31Z",
        "author_association": "MEMBER",
        "body": ">say like int32/int64 and float32/float64 etc\r\n\r\nIs it common to merge on floating-point columns? That sounds inherently dangerous to me, but I suppose it could be on an integer column that was accidentally or necessarily up-casted.",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1176467159/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1176477778",
        "html_url": "https://github.com/rapidsai/cudf/issues/11189#issuecomment-1176477778",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/11189",
        "id": 1176477778,
        "node_id": "IC_kwDOBWUGps5GH6BS",
        "user": {
            "login": "VibhuJawa",
            "id": 4837571,
            "node_id": "MDQ6VXNlcjQ4Mzc1NzE=",
            "avatar_url": "https://avatars.githubusercontent.com/u/4837571?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/VibhuJawa",
            "html_url": "https://github.com/VibhuJawa",
            "followers_url": "https://api.github.com/users/VibhuJawa/followers",
            "following_url": "https://api.github.com/users/VibhuJawa/following{/other_user}",
            "gists_url": "https://api.github.com/users/VibhuJawa/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/VibhuJawa/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/VibhuJawa/subscriptions",
            "organizations_url": "https://api.github.com/users/VibhuJawa/orgs",
            "repos_url": "https://api.github.com/users/VibhuJawa/repos",
            "events_url": "https://api.github.com/users/VibhuJawa/events{/privacy}",
            "received_events_url": "https://api.github.com/users/VibhuJawa/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2022-07-06T17:12:31Z",
        "updated_at": "2022-07-06T17:17:41Z",
        "author_association": "MEMBER",
        "body": "> Is it common to merge on floating-point columns? \r\n\r\nI dont this its common but as you say users can do that accidentally in workflows . A common example i have seen is when people read csvs  and it gets interpreted as different dtypes based on how the data is partitioned.   \r\n\r\nThat said we should have guard rails around it by failing specially in the dask-sql land  (will leave the dask-cudf  behavior to your judgment ) following spark behavior, Like In spark land this would raise something like below: \r\n\r\n```python\r\n Failed to merge incompatible data types IntegerType and DoubleType\r\n ```",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1176477778/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1176485909",
        "html_url": "https://github.com/rapidsai/cudf/issues/11189#issuecomment-1176485909",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/11189",
        "id": 1176485909,
        "node_id": "IC_kwDOBWUGps5GH8AV",
        "user": {
            "login": "ayushdg",
            "id": 19949207,
            "node_id": "MDQ6VXNlcjE5OTQ5MjA3",
            "avatar_url": "https://avatars.githubusercontent.com/u/19949207?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/ayushdg",
            "html_url": "https://github.com/ayushdg",
            "followers_url": "https://api.github.com/users/ayushdg/followers",
            "following_url": "https://api.github.com/users/ayushdg/following{/other_user}",
            "gists_url": "https://api.github.com/users/ayushdg/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/ayushdg/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/ayushdg/subscriptions",
            "organizations_url": "https://api.github.com/users/ayushdg/orgs",
            "repos_url": "https://api.github.com/users/ayushdg/repos",
            "events_url": "https://api.github.com/users/ayushdg/events{/privacy}",
            "received_events_url": "https://api.github.com/users/ayushdg/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2022-07-06T17:21:50Z",
        "updated_at": "2022-07-06T17:23:19Z",
        "author_association": "MEMBER",
        "body": "> In my opinion we should handle merges b/w columns of similar dtype , say like int32/int64 and float32/float64 etc and FAIL loudly in other cases (like int32/float32, int32/uint32 ) for dask-sql .\r\n\r\nAgreed, upcasting integer columns while joining on integer types should be handled by dask-cudf since it's something that works in cudf/pandas and in sql land integer based joins are common and the dtypes may vary. \r\n\r\nDask-sql will also show the same: `UserWarning` for dtypes but if it's a large query, it can be hard to diagnose which part of the query the warning is coming from. \r\n\r\n> Is it common to merge on floating-point columns? That sounds inherently dangerous to me, but I suppose it could be on an integer column that was accidentally or necessarily up-casted.\r\n\r\nEquality joins on float columns are very uncommon and most often happen accidentally. Happy to move that part of the discussion over to dask-sql to improve dtype checking during joins and raising more warnings. This issue can primarily focus on handling integer dtypes. ",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1176485909/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1206711843",
        "html_url": "https://github.com/rapidsai/cudf/issues/11189#issuecomment-1206711843",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/11189",
        "id": 1206711843,
        "node_id": "IC_kwDOBWUGps5H7PYj",
        "user": {
            "login": "github-actions[bot]",
            "id": 41898282,
            "node_id": "MDM6Qm90NDE4OTgyODI=",
            "avatar_url": "https://avatars.githubusercontent.com/in/15368?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/github-actions%5Bbot%5D",
            "html_url": "https://github.com/apps/github-actions",
            "followers_url": "https://api.github.com/users/github-actions%5Bbot%5D/followers",
            "following_url": "https://api.github.com/users/github-actions%5Bbot%5D/following{/other_user}",
            "gists_url": "https://api.github.com/users/github-actions%5Bbot%5D/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/github-actions%5Bbot%5D/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/github-actions%5Bbot%5D/subscriptions",
            "organizations_url": "https://api.github.com/users/github-actions%5Bbot%5D/orgs",
            "repos_url": "https://api.github.com/users/github-actions%5Bbot%5D/repos",
            "events_url": "https://api.github.com/users/github-actions%5Bbot%5D/events{/privacy}",
            "received_events_url": "https://api.github.com/users/github-actions%5Bbot%5D/received_events",
            "type": "Bot",
            "site_admin": false
        },
        "created_at": "2022-08-05T18:03:05Z",
        "updated_at": "2022-08-05T18:03:05Z",
        "author_association": "NONE",
        "body": "This issue has been labeled `inactive-30d` due to no recent activity in the past 30 days. Please close this issue if no further response or action is needed. Otherwise, please respond with a comment indicating any updates or changes to the original issue and/or confirm this issue still needs to be addressed. This issue will be labeled `inactive-90d` if there is no activity in the next 60 days.",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1206711843/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    }
]